crates/ruff_python_parser/src/parser/statement.rs

```diff
@@ -3160,24 +3160,94 @@ impl<'src> Parser<'src> {
         // We should do the same but currently we can't without throwing away the parsed
         // expression because the AST can't contain it.
 
+        // test_ok type_param_type_var_tuple
+        // type X[*Ts] = int
+        // type X[*Ts = int] = int
+        // type X[*Ts = *int] = int
+        // type X[T, *Ts] = int
+        // type X[T, *Ts = int] = int
         if self.eat(TokenKind::Star) {
             let name = self.parse_identifier();
 
+            let default_value = if self.eat(TokenKind::Equal) {
+                if self.at_expr() {
+                    // test_err type_param_type_var_tuple_invalid_default_expr
+                    // type X[*Ts = *int] = int
+                    // type X[*Ts = yield x] = int
+                    // type X[*Ts = yield from x] = int
+                    // type X[*Ts = x := int] = int
+                    Some(Box::new(
+                        self.parse_conditional_expression_or_higher_impl(
+                            ExpressionContext::starred_bitwise_or(),
+                        )
+                        .expr,
+                    ))
```

Comment:
Can you clarify what the grammar is? I'm a bit confused between the two:
1. The reference implementation uses `expression` https://github.com/Gobot1234/cpython/blob/a05c2f31f580bd001dcac34668bb3cbd8d3e0eb0/Grammar/python.gram#L664
2. The PEP specifies two alternatives https://peps.python.org/pep-0696/#grammar-changes:
	```
	type_param_default:
	    | '=' e=expression
	    | '=' e=starred_expression
	```

Although, looking at [Pyright's implementation](https://github.com/microsoft/pyright/blob/f4afe405cdf8d6a95abe67a459e8ea84af4162b5/packages/pyright-internal/src/parser/parser.ts#L574-L582), this does seem correct.

Reply:
In a sense there isn't a grammar yet, because my implementation hasn't been merged into CPython yet. I posted about it here: https://discuss.python.org/t/pep-696-type-defaults-for-typevarlikes/22569/36

Based on the test cases I added, I think I picked the right flavor of expression here to match what I did in CPython.

Reply:
I see, thanks for providing the context. The grammar then matches the implementation.